### DISCUSSION TITLE
Question about prefetch

### DISCUSSION THREAD

**Discussion by SheseR** [ruby-amqp/kicks#9] [General]
Hey all! 
The my idea is to collect 10 messages, and handle those ones in the single process, so I do that like

```
Sneakers.configure  :connection => Bunny.new(...),
                    :workers => 1,
                    :handler => CustomHandler,
                    :prefetch => 10,
                    :threads => 1

class Worker
   from_queue 'some_queue_name
   def work_with_params(deserialized_msg, _delivery_info, _metadata)
        @messages ||= []
        @messages << message.deep_symbolize_keys
        
        return unless @messages.count == 10
  
       # do something with @messages
       @messages = []
       ack!
   end
end

class CustomHandler < Sneakers::Handlers::Maxretry
      def acknowledge(hdr, _props, _msg)
        # add mutliple acknowledge
        @channel.acknowledge(hdr.delivery_tag, true)
      end
end
```

The worker waits for 10 messages before handling. it is work fine,  but if there is no 10 messages in the worker during some time, the server close channel, and worker downs at all
So I wanna to set some 'idle timeout' and catch it on this worker for handling of existed messages. 
Does sneakers support that ? 
thanks a lot! -))

**Comment by michaelklishin** [comment]
@SheseR questions belong to discussions, not issues.

I cannot comment on what happens in your system, you haven't provided a way to reproduce, no logs, no traffic capture, effectively nothing for the maintainers to work with.

What I can confirm is that

 * Kicks supports prefetch because Bunny does, like any other AMQP 0-9-1 client. There is a [dedicated tutorial](https://www.rabbitmq.com/tutorials/tutorial-two-ruby) and a fairly extensive [doc guide](https://www.rabbitmq.com/docs/confirms#channel-qos-prefetch)
 * There is a [consumer delivery timeout](https://www.rabbitmq.com/docs/consumers#acknowledgement-timeout) that protects RabbitMQ nodes, and quorum queues in particular, from not being able to perform compaction when a client gets stuck and does not acknowledge a message for hours or days
 
If you need to process messages strictly in batches, I'd say you should consider using [RabbitMQ streams](https://www.rabbitmq.com/docs/streams) but I am not aware of the state of Ruby clients for that protocol. And then you'd have to give up on using Kicks, too. But it will allow you to consume in batches, and repeatedly if necessary, more predictably than with a queue and a channel prefetch.

**Comment by simi** [comment]
Per my understanding, this is not supported by Kick directly, but I have some code extension to Sneakers somewhere I can probably share to achieve exactly this.
